Try using built in mvn as recommended by AppVeyor support

timothyjward opened the pull request:
According to the AppVeyor support team the source of our slow builds is the maven wrapper that we use. This leads me to believe that AppVeyor have some special settings in their Maven installation which we lose by using the wrapper. This commit switches to using the AppVeyor installed mvn command

Signed-off-by: Tim Ward <tim.ward@paremus.com>

bjhargrave commented:
This would seem to be a bad idea in that we have no control over the version of maven used to build. Also, we use mvnw on the Bnd build without any observed issue. 

In any case, this PR did not remove all of the invocations of mvnw.

timothyjward commented:
> This would seem to be a bad idea in that we have no control over the version of maven used to build. Also, we use mvnw on the Bnd build without any observed issue.

I agree that using `mvnw` is better - that's why I added it to the repo, why it's used by Travis, and why it was used by AppVeyor until this commit. 

The problem here is that mvnw is reliably slow in AppVeyor - so slow that it fails every build by timing out. I've tried the AppVeyor recommended actions at https://www.appveyor.com/docs/how-to/http-proxy/#maven but it made no difference. On contacting AppVeyor support they verified that the builds were slow, and their one recommended solution was to use their built-in mvn command, which does speed up the build to be approximately the same as Travis. They apparently have no special `settings.xml` or other configuration, so there aren't any other actions I can see to take with `mvnw`.

I think this leaves the situation where we either:

* run with `mvn` on AppVeyor, accepting its disadvantages, to keep a level of build testing on Windows. Travis remains the _blessed_ CI server using `mvnw` and publishes its output.

* disable AppVeyor for this project because it won't run with `mvnw`. This eliminates the risk that we see failures due to maven version changes, but means we have no testing on Windows

* revert the commit c75e3f3 as we want to use `mvnw` everywhere, but accept that all of the AppVeyor builds will fail by timing out until some time in the future where the true source of the issue is found by AppVeyor/Sonatype and fixed.

My preferred option is to keep using `mvn` on AppVeyor for now because it means that we get to keep cross-platform testing of the archetypes without sacrificing more time shaving this particular yak. It may be that `mvnw` becomes usable again once we aren't using so many snapshot repositories, at which point I would like to switch back to using it.

bjhargrave commented:
> The problem here is that mvnw is reliably slow in AppVeyor - so slow that it fails every build by timing out.

Strange because we use mvnw on Appveyor in the Bnd builds to build the maven plugins and it does not appear to be "reliably slow" where the builds are failing.
